Packages/com.unity.inputsystem/InputSystem/API/Input.actions.cs

```diff
@@ -0,0 +1,534 @@
+        /// exists on the bindings of this action.
+        /// </summary>
+        /// <typeparam name="TInteraction"></typeparam>
+        /// <typeparam name="TParameter"></typeparam>
+        /// <param name="expr"></param>
+        /// <returns></returns>
+        public TParameter GetInteractionParameter<TInteraction, TParameter>(BindingIndex bindingIndex,
+            Expression<Func<TInteraction, TParameter>> expr)
+            where TInteraction : IInputInteraction
+            where TParameter : struct
+        {
+            Debug.Assert(m_Action != null);
+
+            var result = m_Action.GetParameterValue(expr, m_Action.m_ActionMap.bindings[bindingIndex.ToMapIndex()]);
+            return result ?? default(TParameter);
+        }
+
+        /// <summary>
+        /// Find all interactions of type TInteraction and sets the specified parameter on them.
+        /// </summary>
+        /// <typeparam name="TInteraction"></typeparam>
+        /// <typeparam name="TParameter"></typeparam>
+        /// <param name="expr"></param>
+        /// <param name="value"></param>
+        /// <remarks>
+        /// </remarks>
+        public void SetInteractionParameter<TInteraction, TParameter>(BindingIndex bindingIndex,
+            Expression<Func<TInteraction, TParameter>> expr, TParameter value)
+            where TInteraction : IInputInteraction
+            where TParameter : struct
+        {
+            Debug.Assert(m_Action != null);
+
+            m_Action.ApplyParameterOverride(expr, value, m_Action.m_ActionMap.bindings[bindingIndex.ToMapIndex()]);
+        }
+
+        /// <summary>
+        /// True when the action is pressed.
+        /// </summary>
+        /// <param name="input"></param>
+        /// <remarks>
+        /// </remarks>
+        public static implicit operator bool(Input<TActionType> input)
+        {
+            return input.m_Action.IsPressed();
+        }
+
+        /// <summary>
+        /// Implicitly returns the InputAction inside an Input&lt;TActionType&gt; instance.
+        /// </summary>
+        /// <param name="input"></param>
+        public static implicit operator InputAction(Input<TActionType> input)
+        {
+            return input.action;
+        }
+
+        private unsafe bool WasPhaseChangedThisFrame<TInteraction>(BindingIndex bindingIndex,
+            Func<InputActionState.InteractionState, uint> lastChangedInUpdateGetter)
+            where TInteraction : IInputInteraction
+        {
+            var map = m_Action.GetOrCreateActionMap();
+
+            var bindings = m_Action.bindings;
+            var bindingsCount = bindings.Count;
+            var bindingStartIndexInMap = m_Action.m_BindingsStartIndex;
+
+            if (bindingIndex != BindingIndex.None)
+            {
+                bindingStartIndexInMap = bindingIndex.ToMapIndex();
+                bindingsCount = 1;
+            }
+
+            for (var i = 0; i < bindingsCount; i++)
```

Comment:
Doesn't seem the most efficient solution given spinning these lists on every query call to this function

It does lead me to question the value of the 'simplified' interaction API aspect of this part of the work.
As we haven't really validate this part of the API I would suggest we consider not releasing in 0.8.0-pre